plugins/woocommerce/client/blocks/assets/js/blocks/product-description/edit.tsx

```diff
@@ -173,47 +178,22 @@ function RecursionError() {
 	);
 }
 
-/**
- * Check if block is inside a Query Loop with non-product post type
- *
- * @param {string} clientId The block's client ID
- * @param {string} postType The current post type
- * @return {boolean} Whether the block is in an invalid Query Loop context
- */
-const useIsInvalidQueryLoopContext = ( clientId: string, postType: string ) => {
-	const { getBlockParentsByBlockName } = useSelect( blockEditorStore );
-	const blockParents = useMemo( () => {
-		return getBlockParentsByBlockName( clientId, 'core/post-template' );
-	}, [ getBlockParentsByBlockName, clientId ] );
-
-	return blockParents.length > 0 && postType !== 'product';
-};
-
 export default function ProductDescriptionEdit( {
 	context,
 	__unstableLayoutClassNames: layoutClassNames,
 	__unstableParentLayout: parentLayout,
 	clientId,
-} ) {
+}: ProductDescriptionEditProps ) {
 	const { postId: contextPostId, postType: contextPostType } = context;
 	const hasAlreadyRendered = useHasRecursion( contextPostId );
-
-	const blockProps = useBlockProps();
-	const isInvalidQueryLoopContext = useIsInvalidQueryLoopContext(
-		clientId,
-		contextPostType
-	);
-	if ( isInvalidQueryLoopContext ) {
-		return (
-			<div { ...blockProps }>
-				<Warning>
-					{ __(
-						'The Product Description block requires a product context. When used in a Query Loop, the Query Loop must be configured to display products.',
-						'woocommerce'
-					) }
-				</Warning>
-			</div>
-		);
+	const { hasInvalidContext, warningElement } =
+		useQueryLoopProductContextValidation( {
+			clientId,
+			postType: contextPostType,
+			blockName: 'Product Description',
```

Comment:
Should this be translated?

Reply:
Agreed! Updated in [df30e31](https://github.com/woocommerce/woocommerce/pull/56431/commits/df30e313426936eb9561997494598a3321004bfc).